tests/projects/test_docker_projects.py

```diff
@@ -242,3 +242,46 @@ def test_docker_unknown_uri_artifact_cmd_and_envs():
         "file-plugin://some_path")
     assert cmd == []
     assert envs == {}
+
+
+@pytest.mark.parametrize("volumes, environment, os_environ, expected", [
+    ([], ["VAR1"], {"VAR1": "value1"}, [("-e", "VAR1=value1")]),
+    ([], ["VAR1"], {}, ["should_crash", ("-e", "VAR1=value1")]),
+    ([], ["VAR1"], {"OTHER_VAR": "value1"}, ["should_crash", ("-e", "VAR1=value1")]),
+    (
+        [], ["VAR1", ["VAR2", "value2"]], {"VAR1": "value1"},
+        [("-e", "VAR1=value1"), ("-e", "VAR2=value2")]
+    ),
+    ([], [["VAR2", "value2"]], {"VAR1": "value1"}, [("-e", "VAR2=value2")]),
+    (
+        ["/path:/path"], ["VAR1"], {"VAR1": "value1"},
+        [("-e", "VAR1=value1"), ("-v", "/path:/path")]
+    ),
+    (
+        ["/path:/path"], [["VAR2", "value2"]], {"VAR1": "value1"},
+        [("-e", "VAR2=value2"), ("-v", "/path:/path")]
+    )
+])
+def test_docker_user_specified_env_vars(volumes, environment, expected, os_environ):
+    active_run = mock.MagicMock()
+    run_info = mock.MagicMock()
+    run_info.run_id = "fake_run_id"
+    run_info.experiment_id = "fake_experiment_id"
+    run_info.artifact_uri = "/tmp/mlruns/artifacts"
+    active_run.info = run_info
+    image = mock.MagicMock()
+    image.tags = ["image:tag"]
+
+    if "should_crash" in expected:
+        expected.remove("should_crash")
+        with pytest.raises(KeyError) as excinfo:
```

Comment:
```suggestion
        with pytest.raises(KeyError):
```

To fix the lint error, which you can see in https://travis-ci.org/mlflow/mlflow/jobs/608360341?utm_medium=notification&utm_source=github_status